purchases/src/main/kotlin/com/revenuecat/purchases/common/diagnostics/DiagnosticsSynchronizer.kt

```diff
@@ -91,15 +103,11 @@ internal class DiagnosticsSynchronizer(
     }
 
     private fun getEventsToSync(): List<JSONObject> {
-        val diagnosticsList = diagnosticsFileHelper.readDiagnosticsFile()
-        val diagnosticsInFileCount = diagnosticsList.size
-        if (diagnosticsInFileCount > MAX_NUMBER_EVENTS) {
-            val eventsToRemoveCount = diagnosticsInFileCount - MAX_NUMBER_EVENTS + 1
-            diagnosticsFileHelper.deleteOlderDiagnostics(eventsToRemoveCount)
-            diagnosticsTracker.trackMaxEventsStoredLimitReached(diagnosticsInFileCount, eventsToRemoveCount)
-            return diagnosticsFileHelper.readDiagnosticsFile()
+        var eventsToSync: List<JSONObject> = emptyList()
+        diagnosticsFileHelper.readDiagnosticsFile { stream ->
+            eventsToSync = stream.limit(MAX_EVENTS_TO_SYNC_PER_REQUEST).collect(Collectors.toList())
```

Comment:
Adding a limit to the events in memory/per request was done previously in a separate PR, but I decided to move it to this PR for simplicity